docs/DevGuide/BuildOptimization.md

```diff
@@ -208,6 +208,34 @@ It is recommended to install and use
 to use the full KDE window system) to visualize the output -- the larger graphs
 produced by templight are inefficient to render in the graphviz format.
 
+## Clang profiling
+
+With `clang >= 9`, you can generate a flame graph of what the compiler is
+spending its time on, including specific function instantiations, class
+instantiations, source file parsing, debug info generation, and function
+optimization. To do so, compile your `cpp` of interest with `clang >= 9` and use
+the additional flag `-ftime-trace`. You can find the full compilation command
+for your `cpp` of interest by searching for it in the `compile_commands.json`
+file in your `build` directory. This will produce a
```

Comment:
When doing this myself, I found it easiest to just add the `-ftime-trace` flag to the `-D CMAKE_CXX_FLAGS` cmake option. Then it adds the flags for all executables and you can just run `make ExecutableName` and it will produce the `json` file in the directory where the `.cpp.o` object is stored.

This is also useful when you want to analyze just a single `cpp` file also because all the commands in the `compile_commands` file will already have the `-ftime-trace` flag if you add it to cmake.